concordium-consensus/src/Concordium/TimeMonad.hs

```diff
@@ -22,6 +22,17 @@ class Monad m => TimeMonad m where
     default currentTime :: MonadIO m => m UTCTime
     currentTime = liftIO getCurrentTime
 
+    -- |Measure the time of a given action m a.
+    -- Note. One has to remember to factor in laziness when
+    -- reasoning about the returned 'NominalDiffTime'.
+    measureTime :: m a -> m (NominalDiffTime, a)
+    default measureTime :: m a -> m (NominalDiffTime, a)
```

Comment:
Even if it stays a method, I don't think this default does anything since it is as generic as the signature.